CRM/Civicase/Event/Listener/CaseTypeCategoryIsActiveToggler.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+use CRM_Civicase_Factory_CaseTypeCategoryEventHandler as CaseTypeCategoryEventHandlerFactory;
+use Civi\API\Event\RespondEvent;
+
+/**
+ * Class CRM_Civicase_Event_Listener_CaseTypeCategoryIsActiveToggler.
+ *
+ * Contains extra processing for case type category when it is
+ * enabled / disabled, e.g. show / hide related menu item.
+ */
+class CRM_Civicase_Event_Listener_CaseTypeCategoryIsActiveToggler {
+
+  /**
+   * Update case type category related objects on category update.
+   *
+   * @param \Civi\API\Event\RespondEvent $e
+   *   Event data.
+   */
+  public static function onRespond(RespondEvent $e) {
+    $apiRequest = $e->getApiRequest();
+    if ($apiRequest['entity'] != 'OptionValue' || $apiRequest['action'] != 'create') {
+      return;
+    }
+    if (!self::checkReferrer() || !self::checkOptionGroup($apiRequest['params']['id'])) {
+      return;
+    }
```

Comment:
I think we can move the condition above to a function. Something similar to `shouldRun` or `shouldHandle` ?